src/vice_thumbnails.erl

```diff
@@ -70,31 +70,40 @@ generate(Movie, OutName, Options, Fun) ->
                      Other ->
                        Other
                    end,
-  case bucfile:make_dir(SpritesPath) of
-    ok ->
-      SpritesFiles = filename:join(SpritesPath, "%016d.png"),
-      CmdOptions = [{ref, erlang:make_ref()}, {on, get_stages_number(Sprite)}],
-      case vice:convert(Movie, SpritesFiles, [{output_format, "image2"},
-                                              {video_filtergraph, <<"fps=1/", (bucs:to_binary(Every))/binary, ",scale=", (bucs:to_binary(Width))/binary, ":-1">>},
-                                              {type, video},
-                                              {cmd_options, [{stage, 1}|CmdOptions]}
-                                              |allowed_extensions(Movie)],
-                        {fun ?MODULE:thumbnails_finalize/2, {Movie, OutName, Options, ResultFunOrPid, CmdOptions}}) of
-        {async, Ref} ->
-          case is_pid(ResultFunOrPid) of
-            true ->
-              receive
-                {ok, _In, Out} -> {ok, Out};
-                {error, _In, _Out, Code} -> {error, Code};
-                Unknow -> {error, {invalid_response, Unknow}}
+  case vice:info(Movie, duration, [{type, video}|allowed_extensions(Movie)]) of
+    {ok, Duration} when Duration > Every ->
+      case bucfile:make_dir(SpritesPath) of
+        ok ->
+          SpritesFiles = filename:join(SpritesPath, "%016d.png"),
+          CmdOptions = [{ref, erlang:make_ref()}, {on, get_stages_number(Sprite)}],
+          case vice:convert(Movie, SpritesFiles, [{output_format, "image2"},
+                                                  {video_filtergraph, <<"fps=1/", (bucs:to_binary(Every))/binary, ",scale=", (bucs:to_binary(Width))/binary, ":-1">>},
+                                                  {type, video},
+                                                  {cmd_options, [{stage, 1}|CmdOptions]}
+                                                  |allowed_extensions(Movie)],
+                            {fun ?MODULE:thumbnails_finalize/2, {Duration, OutName, Options, ResultFunOrPid, CmdOptions}}) of
+            {async, Ref} ->
+              case is_pid(ResultFunOrPid) of
+                true ->
+                  receive
+                    {ok, _In, Out} -> {ok, Out};
+                    {error, _In, _Out, Code} -> {error, Code};
+                    Unknow -> {error, {invalid_response, Unknow}}
+                  end;
+                false ->
+                  {async, Ref}
               end;
-            false ->
-              {async, Ref}
+            Error ->
+              Error
           end;
         Error ->
           Error
       end;
+    {ok, Duration} ->
+      error_logger:error_msg("Every parameter is bellow video duration => ~p", [Duration]),
+      error;
```

Comment:
In case or error, we need to return an error tuple.
```suggestion
      {error, duration};
```

Reply:
Added ! https://github.com/G-Corp/vice/pull/13/files#diff-fe7b8882d0a6bc67ae95015c86096a41R104